atomic.h

```diff
@@ -7,22 +7,18 @@
 // Portable atomic operations that works on both CPU and GPU
 // Partly inspired by https://github.com/mbitsnbites/atomic
 
-#if defined(__GNUC__) || defined(__clang__) || defined(__xlc__)
-#define ATOMIC_USE_GCC
-#elif defined(_MSC_VER)
-#define ATOMIC_USE_MSVC
-#include <intrin.h>
-#pragma intrinsic (_InterlockedIncrement)
+#if defined(USE_MSVC_INTRINSICS)
+#include "atomic_msvc.h"
```

Comment:
Since C++11 there is the `<atomic>` header. I don't know if C++11 is already the required version though.

Reply:
It is the required version, however <atomic> wouldn't work since we need to directly modify a memory address allocated by PyTorch/TensorFlow. Only platform-dependent intrinsics can achieve this at the moment.